packages/jellyfish-api-core/__tests__/category/loan/createVault.test.ts

```diff
@@ -0,0 +1,227 @@
+      loanValue: expect.any(String)
+    })
+  })
+
+  it('should createVault with default scheme if CreateVault.loanSchemeId is not given', async () => {
+    const owneraddress = await testing.generateAddress()
+    const vaultId = await testing.rpc.loan.createVault({
+      ownerAddress: owneraddress
+    })
+
+    expect(typeof vaultId).toStrictEqual('string')
+    expect(vaultId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const data = await testing.container.call('getvault', [vaultId])
+    expect(data).toStrictEqual({
+      loanSchemeId: 'default', // Get default loan scheme
+      ownerAddress: expect.any(String),
+      isUnderLiquidation: false,
+      collateralAmounts: [],
+      loanAmount: [],
+      collateralValue: expect.any(String),
+      loanValue: expect.any(String)
+    })
+  })
+
+  it('should not createVault if ownerAddress is invalid', async () => {
+    const promise = testing.rpc.loan.createVault({
+      ownerAddress: '1234'
+    })
+
+    await expect(promise).rejects.toThrow('RpcApiError: \'Error: Invalid owner address\', code: -5, method: createvault')
+  })
+
+  it('should not createVault if loanSchemeId is invalid', async () => {
+    const promise = testing.rpc.loan.createVault({
+      ownerAddress: '',
+      loanSchemeId: '1234'
+    })
+
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test VaultTx execution failed:\nCannot find existing loan scheme with id 1234\', code: -32600, method: createvault')
+  })
+
+  it('should createVault with utxos', async () => {
+    const { txid, vout } = await testing.container.fundAddress(GenesisKeys[0].owner.address, 10)
+    const vaultId = await testing.rpc.loan.createVault({
+      ownerAddress: GenesisKeys[0].owner.address,
+      loanSchemeId: 'scheme'
+    }, [{ txid, vout }])
+    expect(typeof vaultId).toStrictEqual('string')
+    expect(vaultId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const rawtx = await testing.container.call('getrawtransaction', [vaultId, true])
+    expect(rawtx.vin[0].txid).toStrictEqual(txid)
+    expect(rawtx.vin[0].vout).toStrictEqual(vout)
+
+    const data = await testing.container.call('getvault', [vaultId])
+    expect(data).toStrictEqual({
+      loanSchemeId: 'scheme',
+      ownerAddress: GenesisKeys[0].owner.address,
+      isUnderLiquidation: false,
+      collateralAmounts: [],
+      loanAmount: [],
+      collateralValue: expect.any(String),
+      loanValue: expect.any(String)
+    })
+  })
+
+  it('should not createVault with utxos not from the owner', async () => {
+    const utxo = await testing.container.fundAddress(await testing.generateAddress(), 10)
+    const promise = testing.rpc.loan.createVault({
+      ownerAddress: GenesisKeys[0].owner.address,
+      loanSchemeId: 'scheme'
+    }, [utxo])
+
+    const sBuff = new SmartBuffer()
+    OP_CODES.toBuffer(P2PKH.fromAddress(RegTest, GenesisKeys[0].owner.address, P2PKH).getScript().stack, sBuff)
```

Comment:
What is this needed for ?

Reply:
💡 something did for fun.
https://github.com/DeFiCh/ain/blob/a0d49b69cf4907d035419a812f15493e1fa812f9/src/masternodes/mn_checks.cpp#L2063
c++ side returns the hex of the script.

Reply:
But maybe c++ side should return something more straightforward if it hits this error. like the address. But I've seen this in other RPCs as well, it just returns the hex from script, maybe coz it's easy to call `.GetHex()`😛

Reply:
Actually, I have told Diego a few days ago to remove address from the error message and he agreed to me.
If you see other error for `...must have at least one input from token owner...`, they don't have address

Reply:
https://github.com/DeFiCh/ain/issues/731
will update later when the above is updated. not sure what the c++ side wants to do atm.
probably they will remove as per other usages.